packages/ts/react-grid/src/autogrid.tsx

```diff
@@ -77,17 +83,23 @@ function createColumns(model: ModelConstructor<unknown, AbstractModel<unknown>>)
   ));
 }
 
-export function AutoGrid<TItem>({ service, model, ...gridProps }: AutoGridProps<TItem>): JSX.Element {
+export function AutoGrid<TItem>({ service, model, filter, ...gridProps }: AutoGridProps<TItem>): JSX.Element {
   // This cast should go away with #1252
   const children = createColumns(model as ModelConstructor<unknown, AbstractModel<unknown>>);
 
-  return (
-    <Grid
-      {...gridProps}
-      ref={(grid) => {
-        grid!.dataProvider = createDataProvider(grid!, service);
-      }}
-      children={children}
-    ></Grid>
-  );
+  const ref = useRef<GridElement<TItem>>(null);
+  const dataProviderFilter = useRef<Filter | undefined>(undefined);
+
+  useEffect(() => {
+    const grid = ref.current!;
+    grid.dataProvider = createDataProvider(grid, service, dataProviderFilter);
+  });
+
+  useEffect(() => {
```

Comment:
As this file will probably pile up with `useEffect` calls, I would suggest to add a small comment over each on what they do and when they are run.

Reply:
Done